File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/domain/ui/dto/ElementDTO.java
package com.mokatest.platform.demos.domain.ui.dto;

import com.mokatest.platform.demos.domain.ui.Element;
import lombok.Data;

@Data
public class ElementDTO {
    // 选择元素的
    private Element locator;
    // 自定义元素定位的
    private Element customLocator;
    /**
     * 定位来源：LIBRARY=库选元素，CUSTOM=自定义定位。
     * 为空表示历史数据，按「库选优先」规则取值。
     */
    private String locatorSource;
}

File: frontEnd/src/types/dto/ElementDTO.ts
import {Element} from "@/types/domain/Element";

export class ElementDTO {
    // 选择元素的
    locator: Element = new Element();
    // 自定义元素定位的
    customLocator: Element = new Element();
    // 定位来源：LIBRARY=库选元素，CUSTOM=自定义定位；空表示历史数据，后端按「库选优先」规则取值
    locatorSource: string = '';
}


